alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDocLevelMonitorFanOutAction.kt

```diff
@@ -0,0 +1,1186 @@
+                    inputRunResults,
+                    docsToQueries,
+                )
+            }
+            monitorResult = monitorResult.copy(inputResults = InputRunResults(listOf(inputRunResults)))
+
+            /*
+             populate the map queryToDocIds with pairs of <DocLevelQuery object from queries in monitor metadata &
+             list of matched docId from inputRunResults>
+             this fixes the issue of passing id, name, tags fields of DocLevelQuery object correctly to TriggerExpressionParser
+             */
+            queries.forEach {
+                if (inputRunResults.containsKey(it.id)) {
+                    queryToDocIds[it] = inputRunResults[it.id]!!
+                }
+            }
+
+            val idQueryMap: Map<String, DocLevelQuery> = queries.associateBy { it.id }
+
+            val triggerResults = mutableMapOf<String, DocumentLevelTriggerRunResult>()
+            // If there are no triggers defined, we still want to generate findings
+            if (monitor.triggers.isEmpty()) {
+                if (dryrun == false && monitor.id != Monitor.NO_ID) {
+                    createFindings(monitor, docsToQueries, idQueryMap, true)
+                }
+            } else {
+                monitor.triggers.forEach {
+                    triggerResults[it.id] = runForEachDocTrigger(
+                        monitorResult,
+                        it as DocumentLevelTrigger,
+                        monitor,
+                        idQueryMap,
+                        docsToQueries,
+                        queryToDocIds,
+                        dryrun,
+                        executionId = executionId,
+                        workflowRunContext = workflowRunContext
+                    )
+                }
+            }
+
+            // If any error happened during trigger execution, upsert monitor error alert
+            val errorMessage = constructErrorMessageFromTriggerResults(triggerResults = triggerResults)
+            if (errorMessage.isNotEmpty()) {
+                alertService.upsertMonitorErrorAlert(
+                    monitor = monitor,
+                    errorMessage = errorMessage,
+                    executionId = executionId,
+                    workflowRunContext
+                )
+            } else {
+                onSuccessfulMonitorRun(monitor)
+            }
+
+            listener.onResponse(
+                DocLevelMonitorFanOutResponse(
+                    nodeId = clusterService.localNode().id,
+                    executionId = request.executionId,
+                    monitorId = monitor.id,
+                    indexExecutionContext.updatedLastRunContext,
+                    InputRunResults(listOf(inputRunResults)),
+                    triggerResults
+                )
+            )
+        } catch (e: Exception) {
+            log.error("${request.monitor.id} Failed to run fan_out on node ${clusterService.localNode().id} due to error")
```

Comment:
NIT: Log error message and re-write message as follows

`log.error("Doc-Level Monitor ${request.monitor.id} : Failed to run fan_out on node ${clusterService.localNode().id} due to error", e) `

Reply:
plz add the exception to error log